homeassistant/components/twinkly/config_flow.py

```diff
@@ -43,19 +50,61 @@ async def async_step_user(self, user_input=None):
                 await self.async_set_unique_id(device_info[DEV_ID])
                 self._abort_if_unique_id_configured()
 
-                return self.async_create_entry(
-                    title=device_info[DEV_NAME],
-                    data={
-                        CONF_ENTRY_HOST: host,
-                        CONF_ENTRY_ID: device_info[DEV_ID],
-                        CONF_ENTRY_NAME: device_info[DEV_NAME],
-                        CONF_ENTRY_MODEL: device_info[DEV_MODEL],
-                    },
-                )
+                return self._create_entry_from_device(device_info, host)
+
             except (asyncio.TimeoutError, ClientError) as err:
                 _LOGGER.info("Cannot reach Twinkly '%s' (client)", host, exc_info=err)
                 errors[CONF_HOST] = "cannot_connect"
 
         return self.async_show_form(
             step_id="user", data_schema=Schema(schema), errors=errors
         )
+
+    async def async_step_dhcp(
+        self, discovery_info: dhcp.DhcpServiceInfo
+    ) -> data_entry_flow.FlowResult:
+        """Handle dhcp discovery for twinkly."""
+        device_info = await twinkly_client.TwinklyClient(
```

Comment:
```suggestion
        self._async_abort_entries_match({CONF_ENTRY_HOST: discovery_info.ip})
        device_info = await twinkly_client.TwinklyClient(
```

We should above proving the host every time its rediscovered via dhcp if there is already a config entry for that ip as some devices get overloaded easily from discovery.